assets/src/edit-story/app/layout/useZoomSetting.js

```diff
@@ -31,143 +31,178 @@ import {
+      if (workspaceRatio > FULLBLEED_RATIO) {
+        // workspace is limited in the height, so use the width minus room for scrollbar
+        maxPageWidth =
+          workspaceSize.width -
+          themeHelpers.SCROLLBAR_WIDTH -
+          ZOOM_PADDING_LARGE;
+      } else {
+        // workspace is limited in the width, so use the height minus room for scrollbar converted
+        maxPageWidth =
+          (workspaceSize.height - themeHelpers.SCROLLBAR_WIDTH) *
+            FULLBLEED_RATIO -
+          ZOOM_PADDING_LARGE;
       }
-      case ZOOM_SETTING.TRIPLE: {
-        maxPageWidth = 3 * PAGE_WIDTH;
-        break;
+      break;
+    }
+    case ZOOM_SETTING.FIT: {
+      // See how much we can fit inside so all is visible
+      // However, leave some extra space, as we don't want it to hug too tightly to the edges.
+      if (workspaceRatio > FULLBLEED_RATIO) {
+        // workspace is limited in the height, so use the (height - padding) converted
+        maxPageWidth =
+          (workspaceSize.height - ZOOM_PADDING_LARGE) * FULLBLEED_RATIO;
+      } else {
+        // workspace is limited in the width, so use the width - padding
+        maxPageWidth = workspaceSize.width - ZOOM_PADDING_LARGE;
       }
-      default:
-        // do nothing for any of the fixed settings
-        break;
+      break;
     }
-    // Floor page width to nearest multiple of PAGE_WIDTH_FACTOR
-    const pageWidth =
-      PAGE_WIDTH_FACTOR * Math.floor(maxPageWidth / PAGE_WIDTH_FACTOR);
-    // Update whether there's overflow and if we have room for page navigation
-    const pageHeight = pageWidth / PAGE_RATIO;
-    const fullbleedHeight = pageWidth / FULLBLEED_RATIO;
-    // Is full width of available area minus gap used (up to two zoom factors off)
-    const hasHorizontalOverflow =
-      pageWidth >=
-      workspaceSize.width - ZOOM_PADDING_LARGE - 2 * PAGE_WIDTH_FACTOR;
-    // Is full height of available area minus gap used (up to two zoom factors off)
-    const hasVerticalOverflow =
-      pageHeight >=
-      workspaceSize.height - ZOOM_PADDING_LARGE - 2 * PAGE_WIDTH_FACTOR;
-    const hasAnyOverflow = hasHorizontalOverflow || hasVerticalOverflow;
-    const hasPageNavigation =
-      !hasAnyOverflow && pageWidth < workspaceSize.width - 2 * PAGE_NAV_WIDTH;
-    // if any scroll, add a lage padding, else just a bit to allow to show hover frame
-    const pagePadding = hasAnyOverflow ? ZOOM_PADDING_LARGE : ZOOM_PADDING_NONE;
-    // If any kind of scroll, use entire available workspace
-    const viewportWidth = hasAnyOverflow
-      ? workspaceSize.width
-      : pageWidth + pagePadding;
-    const viewportHeight = hasAnyOverflow
-      ? workspaceSize.height
-      : fullbleedHeight + pagePadding;
-    return {
+    case ZOOM_SETTING.SINGLE: {
+      maxPageWidth = PAGE_WIDTH;
+      break;
+    }
+    case ZOOM_SETTING.DOUBLE: {
+      maxPageWidth = 2 * PAGE_WIDTH;
+      break;
+    }
+    case ZOOM_SETTING.TRIPLE: {
```

Comment:
Nit: This can be removed, it seems, doesn't look like the `TRIPLE` is defined nor in use.

Reply:
Thanks, removed in db2be66a0.